test/cql-pytest/conftest.py

```diff
@@ -93,12 +95,30 @@ def this_dc(cql):
 # and automatically deleted at the end. We use scope="session" so that all
 # tests will reuse the same keyspace.
 @pytest.fixture(scope="session")
-def test_keyspace(cql, this_dc):
+def test_keyspace(request, cql, this_dc):
     name = unique_name()
-    cql.execute("CREATE KEYSPACE " + name + " WITH REPLICATION = { 'class' : 'NetworkTopologyStrategy', '" + this_dc + "' : 1 }")
+    replication_options = [("class", "NetworkTopologyStrategy"), (this_dc, 1)]
+    if request.config.getoption("replication") == "tablets":
+        replication_options.append(("initial_tablets", 128))
+    serialized_options = ", ".join([f"'{key}' : '{value}'" for key, value in replication_options])
+    cql.execute("CREATE KEYSPACE " + name + " WITH REPLICATION = { " + serialized_options + " }")
     yield name
     cql.execute("DROP KEYSPACE " + name)
 
+# For tests that don't work with tablets (yet).
```

Comment:
As I commented elsewhere, I think this approach is wrong. We don't have "tests that don't work with tablets" - we have **code** that doesn't work with tablets, causing the test to fail. So we shouldn't have something called "vnode_only", we should have "xfail_on_tablets".

We may indeed need a "tablets_only" for tests that are designed directly to test the tablets feature, but note that I create a couple of days ago (I'm not sure if it went in already?) test/cql-pytest/test_tablets.py, and this file already had its own skip-if-not-tablets function so we don't necessarily need aglobal one in conftest.py

Reply:
Ok, so I'll leave only the vnodes one, and rename it to something else, to better express that this test is disabled temporarily, *until* the bug in the code causing it to fail is fixed. Everything has to support tablets, so something not working with tablets is a bug indeed.

Reply:
Ok, will do.